WordPressCom-Stats-iOS/InsightsTableViewController.m

```diff
@@ -407,6 +904,92 @@ - (void)retrieveStats
              self.sectionData[@(StatsSectionInsightsTodaysStats)] = summary;
          }
      }
+                                              commentsAuthorCompletionHandler:^(StatsGroup *group, NSError *error)
```

Comment:
Yup because each request in the batch takes a different amount of time to complete - some are short some are long. I was thinking about the weak self reference too but I couldn't find any memory leaks with how I have it architected. You don't *always* have to do the weak self thing but I should probably do it to be safe.